federation/cmd/federation-controller-manager/app/options/options.go

```diff
@@ -35,6 +35,10 @@ type ControllerManagerConfiguration struct {
 	Port int `json:"port"`
 	// address is the IP address to serve on (set to 0.0.0.0 for all interfaces).
 	Address string `json:"address"`
+	// federation name.
+	FederationName string `json:"federationName"`
```

Comment:
@nikhiljindal Yes, any name, provided that all federations of which a cluster is a member have unique names.  The federation name does not need to match any DNS configuration parameters, or anything else (that I can think of).